### PR TITLE
Python: Add S3 proxies to PyIceberg Catalog FileIO

### LINKED ISSUE #7951: Python: add proxies as configurable parameter to S3FileSystem
### Feature Request / Improvement

We often need to access S3 endpoints through a proxy, and we could easily add these as configurable parameters for a catalog, similar to s3.endpoint, s3.access-key-id, s3.secret-access-key.

https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#using-proxies

Suggested naming convention:
- s3.proxy.uri

https://github.com/apache/iceberg/blob/master/python/pyiceberg/io/__init__.py#L48

This will be added to s3fs.S3FileSystem as `config_kwargs['proxies'] = dict(scheme=uri)` (https://github.com/fsspec/s3fs/blob/c2d924dfa6e04ad3061714db28d65535d7fe6f02/s3fs/core.py#L233C15-L233C15)

And added to pyarrow.fs.S3FileSystem as proxy_options as a string value.

If this sounds good, I'm happy to take up this issue and contribute to Pyiceberg

### Query engine

None

### PR DESCRIPTION
**Pull request by syun64**
Addresses: https://github.com/apache/iceberg/issues/7951 to allow users to set up botocore proxy configurations at the catalog level

Resolves #7951

### CODE REVIEW

File: python/pyiceberg/io/pyarrow.py [hunk 1]
```diff
@@ -62,7 +62,6 @@
     PyFileSystem,
     S3FileSystem,
 )
-
```
Comment (Fokko) [inline python/pyiceberg/io/pyarrow.py:65]: I think you have to revert this change

File: python/pyiceberg/io/fsspec.py [hunk 2]
```diff
@@ -28,17 +28,18 @@
 from urllib.parse import urlparse
 
 import requests
+
 from botocore import UNSIGNED
 from botocore.awsrequest import AWSRequest
 from fsspec import AbstractFileSystem
 from fsspec.implementations.local import LocalFileSystem
 from requests import HTTPError
-
```
Comment (Fokko) [inline python/pyiceberg/io/fsspec.py:36]: I think you have to revert this change

File: python/pyiceberg/io/fsspec.py [hunk 3]
```diff
@@ -28,17 +28,18 @@
 from urllib.parse import urlparse
 
 import requests
+
```
Comment (Fokko) [inline python/pyiceberg/io/fsspec.py:31]: I think you have to revert this change

**Review by Fokko — APPROVED** [review]
This makes sense to me, thanks for adding this @syun64. Can you add this to the documentation as well? This lives under `python/mkdocs/`. Thanks!

**Comment by syun64** [comment]
> This makes sense to me, thanks for adding this @syun64. Can you add this to the documentation as well? This lives under `python/mkdocs/`. Thanks!

I just made the suggested changes @Fokko - are you be able to merge the PR for me? It looks like only those with write access can

**Comment by syun64** [comment]
> @syun64 It is in, thanks for providing this functionality.
> 
> > Also, it looks like we might have just missed the 0.4.0 release - what is the release cadence for pyiceberg / when are we targeting a 0.4.1 release? We are pretty eager to test this feature to get ourselves fully onboarded to pyiceberg so any guideline regarding the release timeline would be very helpful 👍
> 
> Since 0.4.0 is just out, and I'm seeing quite a few patches, I'm thinking of 0.4.1 somewhere end of this month. Would that help? If you want to test it, you can also install it directly from the repository: https://py.iceberg.apache.org/

Yes, that will be incredibly helpful. We will be testing out the feature by installing it directly from the repository, but of course having the formal release package will allow us to explore more 'Production usecases' as well.

I'll keep tabs on the release schedule / comms to help test the rc when it is ready as well.

Thank you @Fokko !
